backend-shared/config/db/operations_test.go

```diff
@@ -0,0 +1,577 @@
+}
+
+func TestUpdateOperation(t *testing.T) {
+	testSetup(t)
+	defer testTeardown(t)
+	dbq, err := NewUnsafePostgresDBQueries(true, true)
+	if !assert.NoError(t, err) {
+		return
+	}
+	defer dbq.CloseDatabase()
+
+	ctx := context.Background()
+
+	clusterUser := ClusterUser{
+		Clusteruser_id: "test-user-application-1",
+		User_name:      "test-user-application",
+	}
+
+	clusterCredentials := ClusterCredentials{
+		Clustercredentials_cred_id:  "test-cluster-creds-test-1",
+		Host:                        "host",
+		Kube_config:                 "kube-config",
+		Kube_config_context:         "kube-config-context",
+		Serviceaccount_bearer_token: "serviceaccount_bearer_token",
+		Serviceaccount_ns:           "Serviceaccount_ns",
+	}
+
+	managedEnvironment := ManagedEnvironment{
+		Managedenvironment_id: "test-managed-env-1",
+		Clustercredentials_id: clusterCredentials.Clustercredentials_cred_id,
+		Name:                  "my env",
+	}
+
+	gitopsEngineCluster := GitopsEngineCluster{
+		Gitopsenginecluster_id: "test-fake-cluster-1",
+		Clustercredentials_id:  clusterCredentials.Clustercredentials_cred_id,
+	}
+
+	gitopsEngineInstance := GitopsEngineInstance{
+		Gitopsengineinstance_id: "test-fake-engine-instance-id",
+		Namespace_name:          "test-fake-namespace",
+		Namespace_uid:           "test-fake-namespace-1",
+		EngineCluster_id:        gitopsEngineCluster.Gitopsenginecluster_id,
+	}
+
+	operationput := Operation{
+		Operation_id:            "test-operation-1",
+		Instance_id:             gitopsEngineInstance.Gitopsengineinstance_id,
+		Resource_id:             "test-fake-resource-id",
+		Resource_type:           "GitopsEngineInstance",
+		State:                   OperationState_Waiting,
+		Operation_owner_user_id: testClusterUser.Clusteruser_id,
+	}
+
+	err = dbq.CreateClusterUser(ctx, &clusterUser)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	err = dbq.CreateClusterCredentials(ctx, &clusterCredentials)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	err = dbq.CreateManagedEnvironment(ctx, &managedEnvironment)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	err = dbq.CreateGitopsEngineCluster(ctx, &gitopsEngineCluster)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	err = dbq.CreateGitopsEngineInstance(ctx, &gitopsEngineInstance)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	err = dbq.CreateOperation(ctx, &operationput, operationput.Operation_owner_user_id)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	operationget := Operation{
+		Operation_id:            "test-operation-1",
+		Instance_id:             gitopsEngineInstance.Gitopsengineinstance_id,
+		Resource_id:             "test-fake-resource-id",
+		Resource_type:           "GitopsEngineInstance-update",
+		State:                   OperationState_Waiting,
+		Operation_owner_user_id: testClusterUser.Clusteruser_id,
+	}
+
+	err = dbq.GetOperationById(ctx, &operationget)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	err = dbq.UpdateOperation(ctx, &operationget)
```

Comment:
See my comment about update above. It applies here as well.

Reply:
Reopening this from the previous comment that I made on applicationget object, feel free to close it if not necessary^